packages/babel-core/src/config/option-assertions.js

```diff
@@ -0,0 +1,179 @@
+// @flow
+
+import type {
+  IgnoreList,
+  IgnoreItem,
+  PluginList,
+  PluginItem,
+  PluginTarget,
+  SourceMapsOption,
+  SourceTypeOption,
+  CompactOption,
+  RootInputSourceMapOption,
+} from "./options";
+
+export function assertSourceMaps(
+  key: string,
+  value: mixed,
+): SourceMapsOption | void {
+  if (
+    value !== undefined &&
+    typeof value !== "boolean" &&
+    value !== "inline" &&
+    value !== "both"
+  ) {
```

Comment:
I wonder if these would be more readable if you applied De Morgan (`!(value === undefined || ...)`)

Reply:
I usually try to keep `return`s at the end, so I ended up going this route. I don't feel like it makes a huge difference.